plugins/language_tal.lua

```diff
@@ -1,29 +1,28 @@
 --mod-version:3
-local syntax = require 'core.syntax'
 
 local label, sublabel = "function", "keyword2"
 
-syntax.add {
+require 'core.syntax'.add {
 	name = "Uxntal",
 	files = { "%.tal$" },
 	block_comment = { '(', ')' },
 	patterns = {
-		{ pattern = {'%(', '%)'},       type = "comment"  },
-		{ pattern = "@%S*%s+%f[[]",     type = label      },
-		{ pattern = "@%S+",             type = "string"   },
-		{ pattern = "%u+()[2kr]*",      type = { "symbol", "keyword" } },
-		{ pattern = "%%%S+",            type = "keyword"  },
-		{ pattern = "&%S+",             type = sublabel   },
-		{ pattern = "\"%S+",            type = "string"   },
-		{ pattern = "%.%S+()/%S*",      type = { label, sublabel } },
-		{ pattern = "%.%S+",            type = label      },
-		{ pattern = "|%x+",             type = "string"   },
-		{ pattern = "[.,;_=-]%S+",      type = sublabel   },
-		{ pattern = "%$%d+",            type = "number"   },
-		{ pattern = "#?%x%x%x%x%f[%X]", type = "number"   },
-		{ pattern = "#?%x%x%f[%X]",     type = "number"   },
-		{ pattern = "[!?]()[^%[%]{}%s]+", type = { "operator", "function" } },
-		{ pattern = "[^%[%]{}%s]+",     type = "function" },
+		{ pattern = { "%f[%(]%(%s", "%)" }, type = "comment"  },
```

Comment:
What's the reason for this change?

Reply:
comment delimiters must have at least a space between them and any other character to be recognized as comments, that's just an easier way to notice when you forget to put space in at least the left side, because I don't quite know how to catch the right side too :/

Reply:
Wait, could you make some examples of valid and invalid comments?

Reply:
- `(invalid)`, `(invalid )`, `( invalid)`
- `( valid )`, `(     valid   )`, `( valid        )`
at least according to uxnasm and Drifblim

Reply:
No requirements about what comes before `(` and after `)`?

Reply:
![IMG_20231210_092811.jpg](https://github.com/lite-xl/lite-xl-plugins/assets/70547062/a1ec7bf7-06f3-40e7-8d3f-11ff2b599abc)

the extra spaces won't show up :(

Reply:
Looks fine here:
![image](https://github.com/lite-xl/lite-xl-plugins/assets/2798487/921ee6f9-6813-47e8-bab5-6d4c7fb582f6)

Reply:
If there aren't requirements before/after the comment, wouldn't `{ "%(%s", "%s%)" }` work fine?

Reply:
> No requirements about what comes before `(` and after `)`?

there can't be any character before or after the `(`, and the right hand side only closes when the `)` is alone

as in, `( test) some more stuff` will eat everything after the `)`

Reply:
I guess you can either make this a regex, or split it into multiple patterns that match the various cases.